sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/SimpleAccessorNameNotationCheck.java

```diff
@@ -242,10 +241,7 @@ private static String getNameOfSettingField(DetailAST assign,
             else {
                 if (assigningFirstChild.getType() == TokenTypes.DOT) {
 
-                    if (assigningFirstChild.getChildCount() == 2
-                            && "this".equals(assigningFirstChild
-                                    .getFirstChild().getText())
-                            && assigningFirstChild.getLastChild().getType() == TokenTypes.IDENT) {
```

Comment:
`this` can only be followed by field name or method name, both of which are `IDENT`, right?